### PR TITLE
feat(Tooltip): only text simplified version

### PR DESCRIPTION
**Pull request by imoutaharik**
Simplified version of the Tooltip.

Some do and don'ts:
- The Trigger is a button and should only accept  an element like Icons or Text.
- The component doesn't use the Popper component to keep it as simple as possible. (Popper may be revisited in a separate PR)
- Use case like Tooltip wrapping a Dialog Trigger button will be handle in a separate PR.
- The Tooltip should be used only for small Text, No interactive elements allowed. (Use Popover for that)

Most simple usage :
```
<Tooltip content="Your Text Here">
   <IconIcon24 />
 </Tooltip>

### CODE REVIEW

File: src/components/Tooltip/Tooltip.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    openOnMount: true,
+};
+export const WithLineBreak = Template.bind({});
+WithLineBreak.args = {
+    content: 'This is a<br>sample text<br />with line breaks\nin it',
+};
+export const Disabled = Template.bind({});
+Disabled.args = {
+    disabled: true,
+};
```
[the remark this answers is not in the document]
Comment (imoutaharik) [reply src/components/Tooltip/Tooltip.stories.tsx:120]: Haha sorry, always the same thing. I'll keep that in mind. That usage note from the code editor always give me the illusion there are actual spaces in there 🙈 
Screenshot 2023-10-13 at 08 53 07

File: src/components/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    });
+    const currentPlacement = state?.placement ?? placement;
+    const arrowStyling = getArrowClasses(currentPlacement);
+    const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+    const handleHideTooltip = useCallback(() => {
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+        timeoutRef.current = setTimeout(() => setOpen(false), leaveDelay);
```
[the remark this answers is not in the document]
Comment (imoutaharik) [reply src/components/Tooltip/Tooltip.tsx:98]: Updated.

File: src/components/Popper/Popper.tsx [hunk 3]
```diff
@@ -99,18 +69,19 @@ export const Popper = ({
...
 
-    const value = useMemo(() => ({ open }), [open]);
     return (
-        <PopperContext.Provider value={value}>
+        <>
             {Children.map(children, (child) => {
                 if (isValidElement(child) && typeof child.type === 'function') {
-                    const { name } = child.type;
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore Property 'displayName' does not exist on type 'JSXElementConstructor<any>'.ts(2339)
```
[the remark this answers is not in the document]
Comment (ryancarville) [reply src/components/Popper/Popper.tsx:77]: @SamuelAlev,

@imoutaharik has gone on her holidays now for 2 weeks.  But I do see the exact same logic in the `Overlays` component and I know she had to encapsulate the `Popper` logic for the `Tooltip` for simplicity.

[the remark this answers is not in the document]
Comment (imoutaharik) [reply src/components/Popper/Popper.tsx:77]: This shouldn't be part of this PR, I forgot to clean, I was testing the logic of the Overlay in this one. I'll revert it that change now and just keep removing the Context.
I'm going to rework the Popper component and test it on a separate PR when I'm back. I don't want to block Ryan as he needs the Tooltip.

But to answer your question, we have some issues with the name. It gets minimized somehow when used on the web-app and the Popper logic wasn't working at all.  I plan to revisit it. But I'll take my time when back from holidays.

File: src/components/Tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    const [arrowElement, setArrowElement] = useState<HTMLDivElement | null>(null);
+    const { styles, attributes, state } = usePopper(referenceElement, popperElement, {
+        placement,
+        modifiers: [
+            { name: 'arrow', options: { element: arrowElement } },
+            { name: 'offset', options: { offset } },
+            {
+                name: 'flip',
+                enabled: flip,
+            },
```
Comment (SamuelAlev) [inline src/components/Tooltip/Tooltip.tsx:86]:
```suggestion
            { name: 'flip', enabled: flip },
```

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**